spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -629,6 +629,38 @@
         end
       end
     end
+
+    def stub_auth
+      allow(controller).to receive(:validate_saml_request).and_return(true)
+      allow(controller).to receive(:verify_authn_context).and_return(true)
+      allow(controller).to receive(:confirm_two_factor_authenticated).and_return(true)
+      allow(controller).to receive(:link_identity_from_session_data).and_return(true)
+    end
+
+    context 'user is redirected to IdV' do
+      it 'tracks the authentication and IdV redirection event' do
+        stub_analytics
+        stub_auth
+        allow(controller).to receive(:identity_needs_verification?).and_return(true)
+
+        expect(@analytics).to receive(:track_event).with('SAML Auth + IdV: true')
```

Comment:
Message text could be better here I think.

`SAM Auth (verification required)`   and `SAM Auth (no verification)`

or 

`SAM Auth (needs_idv=true)`   and `SAM Auth (needs_idv=false)`

Or just put brackets around the `IdV: true`?